File: skills/desktop-app-dev/clients/MediaClient.swift
import Foundation

struct MediaClient {
    let baseURL: URL
    let token: String?

    init(baseURL: URL = URL(string: "http://127.0.0.1:8765")!, token: String? = nil) {
        self.baseURL = baseURL
        self.token = token
    }

    private func authorize(_ request: inout URLRequest) {
        if let token {
            request.setValue("Bearer \(token)", forHTTPHeaderField: "Authorization")
        }
    }

    func enqueue(
        kind: String,
        payload: [String: Any],
        dedupeKey: String? = nil
    ) async throws -> Data {
        var request = URLRequest(url: baseURL.appendingPathComponent("tasks"))
        request.httpMethod = "POST"
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
        authorize(&request)
        var body: [String: Any] = ["kind": kind, "payload": payload]
        if let dedupeKey {
            body["dedupe_key"] = dedupeKey
        }
        request.httpBody = try JSONSerialization.data(withJSONObject: body)
        let (data, _) = try await URLSession.shared.data(for: request)
        return data
    }

    func task(id: Int) async throws -> Data {
        let url = baseURL.appendingPathComponent("tasks/\(id)")
        var request = URLRequest(url: url)
        authorize(&request)
        let (data, _) = try await URLSession.shared.data(for: request)
        return data
    }

    func depsProgress() async throws -> Data {
        let url = baseURL.appendingPathComponent("deps/progress")
        var request = URLRequest(url: url)
        authorize(&request)
        let (data, _) = try await URLSession.shared.data(for: request)
        return data
    }

    func depsStatus() async throws -> Data {
        let url = baseURL.appendingPathComponent("deps/status")
        var request = URLRequest(url: url)
        authorize(&request)
        let (data, _) = try await URLSession.shared.data(for: request)
        return data
    }

    func installDeps() async throws -> Data {
        let url = baseURL.appendingPathComponent("deps/install")
        var request = URLRequest(url: url)
        request.httpMethod = "POST"
        authorize(&request)
        let (data, _) = try await URLSession.shared.data(for: request)
        return data
    }
}

File: skills/desktop-app-dev/clients/MediaClient.kt
import okhttp3.MediaType.Companion.toMediaType
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody.Companion.toRequestBody
import org.json.JSONObject

class MediaClient(
    private val baseUrl: String = "http://127.0.0.1:8765",
    private val token: String? = null,
) {
    private val client = OkHttpClient()

    private fun builder(path: String): Request.Builder {
        val requestBuilder = Request.Builder().url("$baseUrl$path")
        if (token != null) {
            requestBuilder.header("Authorization", "Bearer $token")
        }
        return requestBuilder
    }

    fun enqueue(kind: String, payload: Map<String, Any?>, dedupeKey: String? = null): String {
        val body = mapOf("kind" to kind, "payload" to payload, "dedupe_key" to dedupeKey)
        val json = JSONObject(body).toString()
        val request = builder("/tasks")
            .post(json.toRequestBody("application/json".toMediaType()))
            .build()
        return client.newCall(request).execute().body!!.string()
    }

    fun task(id: Long): String {
        val request = builder("/tasks/$id").build()
        return client.newCall(request).execute().body!!.string()
    }

    fun taskProgress(id: Long): String {
        val request = builder("/tasks/$id/progress").build()
        return client.newCall(request).execute().body!!.string()
    }

    fun taskEvents(id: Long, after: Int = 0, timeout: Double = 0.0): String {
        val query = if (timeout > 0) "after=$after&timeout=$timeout" else "after=$after"
        val request = builder("/tasks/$id/events?$query").build()
        return client.newCall(request).execute().body!!.string()
    }

    fun depsProgress(): String {
        val request = builder("/deps/progress").build()
        return client.newCall(request).execute().body!!.string()
    }

    fun depsStatus(): String {
        val request = builder("/deps/status").build()
        return client.newCall(request).execute().body!!.string()
    }

    fun formats(): String {
        val request = builder("/formats").build()
        return client.newCall(request).execute().body!!.string()
    }

    fun installDeps(): String {
        val request = builder("/deps/install")
            .post("{}".toRequestBody("application/json".toMediaType()))
            .build()
        return client.newCall(request).execute().body!!.string()
    }
}
